File: desafio-02/robsonalexandre/python/primos.py
#!/usr/bin/env python3
"""
Desafio 02: Listar números primos entre 1 e 10000
"""
import math
for num in range(2, 10001):
    md = int(math.sqrt(num))
    for div in range(2, md+1):
        if num % div == 0:
            break
    else:
        print(num)

File: desafio-02/miguelvitta/c/primos.c
#include <stdio.h>
#include <math.h>

int main()
{
    printf("2\n");
    for (int i = 3; i < 10000; i++)
    {
        int notP = 0;
        if(i % 2 != 0)
        {
            int j = 2;
            int result = 1;
            while(notP == 0 && j <= sqrt(i))
            {
                result = i % j;
                if(result == 0)
                {
                    notP++;
                }
                j++;
            }
            if(notP == 0)
            {
                printf("%d\n", i);
            }
        }
    }

    return 0;
}